allensdk/brain_observatory/behavior/data_files/_data_file_abc.py

```diff
@@ -0,0 +1,92 @@
+import abc
+from typing import Any, Union
+from pathlib import Path
+
+
+class DataFile(abc.ABC):
+    """An abstract class that prototypes methods for accessing internal
+    data files.
+
+    These data files contain information necessary to sucessfully instantiate
+    one or many `DataObject`(s).
+
+    External users should ignore this class (and subclasses) as as they
+    will only ever be using `from_nwb()` and `to_nwb()` `DataObject` methods.
+    """
+
+    def __init__(self, filepath: Union[str, Path]):  # pragma: no cover
+        self._filepath: str = str(filepath)
+        self._data = self.load_data(filepath=filepath)
+
+    @property
+    def data(self) -> Any:  # pragma: no cover
+        return self._data
+
+    @property
+    def filepath(self) -> str:  # pragma: no cover
+        return self._filepath
+
+    @classmethod
+    @abc.abstractmethod
+    def from_json(cls) -> "DataFile":  # pragma: no cover
+        """Populates a DataFile from an input *.json likely parsed by
```

Comment:
this now seems confusing to me.
Shouldn't this say "populates a DataFile from a JSON-compatible dictionary"
?

Reply:
I've updated the docstrings to hopefully be more clear